Add documentation for trafikverket_train (#9333)
home-assistant/home-assistant.github.io: source/_components/trafikverket_train.markdown

---
layout: page
title: "Trafikverket Train"
description: "Instructions how to integrate Trafikverket Train within Home Assistant."
date: 2019-04-27 16:32
sidebar: true
comments: false
sharing: true
footer: true
logo: trafikverket.png
ha_category:
  - Transport
  - Sensor
ha_release: 0.93
ha_iot_class: Cloud Polling
---

Retreive train departure information from [Trafikverket](https://www.trafikverket.se/).

Use cases:

- Retreive the next departure and information connected to it for a specific train line.
- Retreive information for a specific departure time on a specific train line.
- Set up alert or perform actions if your train is delayed or cancelled.

Data that is retreived: 

- Next departure for specific train line.
- Canceled status.
- Number of minutes delayed.
- Planned time if no delays occur.
- Estimated time of arrival if delays occur.
- Actual time - when it did arrive.
- Other information / additional texts.
- Deviations.

## {% linkable_title Configuration %}

To enable this sensor, use the following format in your `configuration.yaml`.

```yaml

> sensor:

Reviewer: Only required items in the first example, `weekday` should not be included in this example. You can add it to your example further below.

Author: Done

---
layout: page
title: "Trafikverket Train"
description: "Instructions how to integrate Trafikverket Train within Home Assistant."
date: 2019-04-27 16:32
sidebar: true
comments: false
sharing: true
footer: true
logo: trafikverket.png
ha_category:
  - Transport
  - Sensor
ha_release: 0.96
ha_iot_class: Cloud Polling
---

> Retreive train departure information from [Trafikverket](https://www.trafikverket.se/).

Reviewer 2: ```suggestion
Retrieve train departure information from [Trafikverket](https://www.trafikverket.se/).
```

Use cases:

> - Retreive the next departure and information connected to it for a specific train line.

Reviewer 2: ```suggestion
- Retrieve the next departure and information connected to it for a specific train line.
```

> - Retreive information for a specific departure time on a specific train line.

Reviewer 2: ```suggestion
- Retrieve information for specific departure time on a specific train line.
```

> - Set up alert or perform actions if your train is delayed or cancelled.

Reviewer 2: ```suggestion
- Set up an alert or perform actions if your train is delayed or canceled.
```

> Data that is retreived: 

Reviewer 2: ```suggestion
Data that is retrieved: 
```

> - Next departure for specific train line.

Reviewer 2: ```suggestion
- Next departure for the specific train line.
```

- Canceled status.

> - Number of minutes delayed.

Reviewer 2: ```suggestion
- The number of minutes delayed.
```

- Planned time if no delays occur.
- Estimated time of arrival if delays occur.
- Actual time - when it did arrive.
- Other information / additional texts.
- Deviations.

## {% linkable_title Configuration %}

To enable this sensor, use the following format in your `configuration.yaml`.

```yaml
sensor:
   - platform: trafikverket_train

>      api_key: !secret trafikverket_api

Reviewer: ```suggestion
     api_key: TRAFIKVERKET_API
```

     trains:
     - name: "Morning train to Malmö"
       from: "Stockholm Central"
       to: "Malmö C"
       time: "9:25"
       
     - name: "Next train to Uppsala"
       from: "Stockholm Central"
       to: "Uppsala C"
```

{% configuration %}
api_key:
  description: Your personal API key from Trafikverket.
  required: true
  type: string
trains:
  description: Specify details on the departures to monitor.
  required: true
  type: map
  keys:
    name:
      description: The name of the departure, will be the sensor name.
      required: true
      type: string
    from:
      description: The station from where the train departs.
      required: true
      type: string
    to:
      description: The destination station.
      required: true
      type: string
    time:

>       description: Departure time to monitor, if not entered it will retreive the next departure.

Reviewer 2: ```suggestion
      description: Departure time to monitor, if not entered it will retrieve the next departure.
```

>       required: true

Reviewer 3: The time option isn't required, right?

Reviewer 2: Correct. Adjusted.

      type: string
    weekday:
      description: Specify which days in the week to monitor the specific departure.
      required: false
      type: list
      default: "[mon, tue, wed, thu, fri, sat, sun]"
{% endconfiguration %}

## {% linkable_title Obtaining API key %}

Please click [here](https://api.trafikinfo.trafikverket.se/) and register to obtain the API key.

## {% linkable_title Train station names %}

> Click [here](https://www.trafikverket.se/trafikinformation/tag/?ArrDep=departure&) to see example of train station names.

Reviewer 2: ```suggestion
Click [here](https://www.trafikverket.se/trafikinformation/tag/?ArrDep=departure&) to see an example of train station names.
```

---
layout: page
title: "Trafikverket Train"
description: "Instructions how to integrate Trafikverket Train within Home Assistant."
date: 2019-04-27 16:32
sidebar: true
comments: false
sharing: true
footer: true
logo: trafikverket.png

> ha_category: Transport

Reviewer: ```suggestion
ha_category:
  - Transport
  - Sensor
```

Author: Done, thanks! :)

ha_release: 0.93
ha_iot_class: Cloud Polling

> redirect_from:

Reviewer: Only for already existing pages.
